app/decorators/offence_decorator.rb

```diff
@@ -32,6 +39,14 @@ def plea_sentence(plea)
   end
 
   def mode_of_trial_reason_descriptions
-    mode_of_trial_reasons.map { |reason| reason.description || t('generic.not_available') }
+    mode_of_trial_reasons.map { |reason| mode_of_trial_reason_description(reason) }
+  end
+
+  def mode_of_trial_reason_description(reason)
+    if MODE_OF_TRIAL_REASON_MAPPINGS.key?(reason.code.to_i)
+      MODE_OF_TRIAL_REASON_MAPPINGS[reason.code.to_i]
+    else
+      reason.description || t('generic.not_available')
+    end
```

Comment:
You may be able to replace this with:

```ruby
MODE_OF_TRIAL_REASON_MAPPINGS[reason.code.to_i] || reason.description || t('generic.not_available')
```

(but I haven't checked).

Reply:
or possibly 
```
MODE_OF_TRIAL_REASON_MAPPINGS.fetch(reason.code.to_i, reason.description) || t('generic.not_available')
```

Reply:
Neither of these suggestions quite worked but have amended to `MODE_OF_TRIAL_REASON_MAPPINGS.fetch(reason.code.to_i, reason.description || t('generic.not_available'))` which is similar.